src/main/java/com/o19s/es/ltr/feature/store/FeatureSupplier.java

```diff
@@ -0,0 +1,126 @@
+package com.o19s.es.ltr.feature.store;
+
+import com.o19s.es.ltr.feature.FeatureSet;
+import com.o19s.es.ltr.ranker.LtrRanker;
+
+import java.util.AbstractMap;
+import java.util.AbstractSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.function.Supplier;
+
+
+public class FeatureSupplier extends AbstractMap<String, Float> implements Supplier<LtrRanker.FeatureVector> {
+    private Supplier<LtrRanker.FeatureVector> vectorSupplier;
+    private final FeatureSet featureSet;
+
+    FeatureSupplier(FeatureSet featureSet) {
+        this.featureSet = featureSet;
+    }
+
+    @Override
+    public LtrRanker.FeatureVector get() {
+        return vectorSupplier.get();
+    }
+
+    public void set(Supplier<LtrRanker.FeatureVector> supplier) {
+        this.vectorSupplier = supplier;
+    }
+
+    /**
+     * Returns {@code true} if this map contains a mapping for the specified
+     * featureName.
+     *
+     * @param featureName featureName whose presence in this map is to be tested
+     * @return {@code true} if this map contains a mapping for the specified
+     * featureName
+     * @throws ClassCastException if the key is of an inappropriate type for
+     *                            this map
+     */
+    @Override
+    public boolean containsKey(Object featureName) {
+        try {
+            featureSet.featureOrdinal((String) featureName);
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns the score to which the specified featureName is mapped,
+     * or {@code null} if this map contains no mapping for the featureName.
+     *
+     * @param featureName the featureName whose associated score is to be returned
+     * @return the score to which the specified featureName is mapped, or
+     * {@code null} if this map contains no mapping for the key
+     */
+    @Override
+    public Float get(Object featureName) {
+        int featureOrdinal;
+        try {
+            featureOrdinal = featureSet.featureOrdinal((String) featureName);
+        } catch (IllegalArgumentException e) {
```

Comment:
I'd let IllegalArgumentException propagate, it's likely that it'll endup with an obscure NullPointerException in the end. I think it's preferable to see "Unknown feature XYZ".